### PR TITLE
Fix links and add Java to prerequisites in README

### PR DESCRIPTION
**Pull request by mraible**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by mraible** [comment]
@jhipster/developers Does anyone know why we [prohibit the use of "Java"](https://github.com/jhipster/generator-jhipster/blob/main/test/support/check-enforcements.ts#L62) in the generated README?

It seems like we should specify Java alongside Node.js as a dependency.

**Comment by mshima** [comment]
> JHipster was originally created to allow Java developers to not have Node installed, so it'd be cool to just have Java required

Node is only required to generate the project. We have `./npmw` if node is required for something.

For example:
`./npmw start`

**Comment by mraible** [comment]
Right, but Java most Java developers have an allergic reaction to Node. 😉

That's why I want to emphasize the commands they're familiar with first.

File: generators/client/templates/README.md.jhi.client.ejs [hunk 1]
```diff
@@ -54,6 +54,8 @@ You will only need to run this command when dependencies change in [package.json
 
 We use <%= clientPackageManager %> scripts and <&- fragments.customCliSection(' ') &>[Webpack][] as our build system.
 
+2. [Java](https://dev.java/): Java is used to compile and run the backend Spring Boot server.
+
```
Comment (mshima) [inline generators/client/templates/README.md.jhi.client.ejs:58]:
```suggestion
```

**Pull request merged**